craft_application/commands/init.py

```diff
@@ -0,0 +1,144 @@
+        )
+        parser.add_argument(
+            "--name",
+            type=str,
+            default=None,
+            help="The name of project; defaults to the name of <project_dir>",
+        )
+        # TODO: this fails to render in `--help` (#530)
+        parser.add_argument(
+            "--profile",
+            type=str,
+            choices=self.profiles,
+            default=self.default_profile,
+            help=(
+                f"Use the specified project profile (default is {self.default_profile}, "
+                f"choices are {humanize_list(self.profiles, 'and')})"
+            ),
+        )
+
+    @property
+    def parent_template_dir(self) -> pathlib.Path:
+        """Return the path to the directory that contains all templates."""
+        with importlib.resources.path(
+            self._app.name, "templates"
+        ) as _parent_template_dir:
+            return _parent_template_dir
+
+    @property
+    def profiles(self) -> list[str]:
+        """A list of profile names generated from template directories."""
+        template_dirs = [
+            path for path in self.parent_template_dir.iterdir() if path.is_dir()
+        ]
+        return sorted([template.name for template in template_dirs])
+
+    def run(self, parsed_args: argparse.Namespace) -> None:
+        """Run the command."""
+        project_dir = self._get_project_dir(parsed_args)
+        project_name = self._get_name(parsed_args)
+        template_dir = pathlib.Path(self.parent_template_dir / parsed_args.profile)
+
+        craft_cli.emit.progress("Checking project directory.")
+        self._services.init.check_for_existing_files(
+            project_dir=project_dir, template_dir=template_dir
+        )
+
+        craft_cli.emit.progress("Initialising project.")
+        self._services.init.initialise_project(
+            project_dir=project_dir,
+            project_name=project_name,
+            template_dir=template_dir,
+        )
+        craft_cli.emit.message("Successfully initialised project.")
+
+    def _get_name(self, parsed_args: argparse.Namespace) -> str:
+        """Get name of the package that is about to be initialised.
+
+        Check if name is set explicitly or fallback to project_dir.
+        """
+        if parsed_args.name is not None:
+            return cast(str, parsed_args.name)
+        return self._get_project_dir(parsed_args).name
+
+    @staticmethod
+    def _get_project_dir(parsed_args: argparse.Namespace) -> pathlib.Path:
+        """Get project dir where project should be initialised.
+
+        It applies rules in the following order:
+        - if <project_dir> is specified explicitly, it returns <project_dir>
+        - if <project_dir> is undefined, it defaults to cwd
+        """
+        # if set explicitly, just return it
+        if parsed_args.project_dir is not None:
+            return pathlib.Path(parsed_args.project_dir)
```

Comment:
We should probably convert this into an absolute path:

```suggestion
            return pathlib.Path(parsed_args.project_dir).expanduser().resolve()
```